### PR TITLE
Add volume to change shared memory limit on notebook

### PR DESCRIPTION
**Pull request by harshad16**
Related-to: https://github.com/opendatahub-io/kubeflow/pull/110
Closes: https://github.com/opendatahub-io/odh-dashboard/issues/1593

## Description

This includes a volumeMount onto each Notebook on creation. 
which would allow user to have shard memory based on their node size.

## How Has This Been Tested?

- Deploy ODH or RHODS with odh-deployer switch with this PR image.
`quay.io/opendatahub/odh-dashboard:pr-1594`
- Create a Notebook via jupyter tile or project workbenches
- Check if the notebook is set up with the required volume mount.

Screenshot from 2023-07-27 12-37-20

## Test Impact

Tested this behavior by creating notebook both on jupyter tile and project notebook.
Not sure, how to include test, if some one can provide example. 
the code change would be caught by an end to end test.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by andrewballantyne** [comment]
Please add test steps before taking WIP off

**Comment by harshad16** [comment]
This is Ready for Review.
We would like this in 1.31 release, please try to get this in.
cc: @Gkrumbach07 @alexcreasy

**Review by Gkrumbach07 — APPROVED** [review]
/lgtm

**Comment by harshad16** [comment]
@Gkrumbach07 @alexcreasy  Re-pinging for review
I would like this in 1.31 feature freeze.
apologies for short notice.

File: frontend/src/api/k8s/notebooks.ts
Comment (DaoDaoNoCode) [inline frontend/src/api/k8s/notebooks.ts:85]: This is problematic, when I update the notebook storage, it will duplicated add the memory volumes and volumeMounts

Comment (andrewballantyne) [reply frontend/src/api/k8s/notebooks.ts:85]: Nice catch!

Comment (DaoDaoNoCode) [reply frontend/src/api/k8s/notebooks.ts:85]: So I guess the Elyra ones could have the same problem?

Comment (andrewballantyne) [reply frontend/src/api/k8s/notebooks.ts:85]: cc @harshad16

Comment (harshad16) [reply frontend/src/api/k8s/notebooks.ts:85]: I will try to fix it,
if you have any suggestions on how to fix it, please do share.

Comment (harshad16) [reply frontend/src/api/k8s/notebooks.ts]: Made the changes as suggested.

Thanks @DaoDaoNoCode :100:

**Comment by harshad16** [comment]
Ready for Re-Review

**Comment by harshad16** [comment]
Rebase with main and pushed :+1:

**Review by DaoDaoNoCode — APPROVED** [review]
Tested, works as expected, LGTM

**Comment by harshad16** [comment]
Thanks @DaoDaoNoCode @Gkrumbach07 
@andrewballantyne needs your approval for this to be merged.

**Comment by andrewballantyne** [comment]
No it does not, despite what the bot says. We have a two-approver policy in the Dashboard team. I'm just the only one with standard approval powers.

[Our Process](https://docs.google.com/document/d/1-q2PWqqZc-607h__bk1gFDF4InrB-M9E9rSUBSzDU9w/edit#heading=h.bz8avvor9rse)

**Comment by harshad16** [comment]
ack, @Gkrumbach07 or @DaoDaoNoCode can one of you add the approver label to get this near to finish line?

**Pull request merged**
